File: maava_restaurant/lib/features/orders/domain/order_model.dart
class OrderItemModel {
  OrderItemModel({
    required this.name,
    required this.variantName,
    required this.quantity,
    required this.price,
    required this.isVeg,
    required this.notes,
  });

  factory OrderItemModel.fromJson(Map<String, dynamic> json) {
    num? asNum(dynamic v) => v is num ? v : num.tryParse((v ?? '').toString());
    return OrderItemModel(
      name: (json['name'] ?? '').toString(),
      variantName: (json['variantName'] ?? '').toString(),
      quantity: (asNum(json['quantity']) ?? 1).toInt(),
      price: asNum(json['price'])?.toDouble() ?? 0,
      isVeg: json['isVeg'] == true,
      notes: (json['notes'] ?? '').toString(),
    );
  }

  final String name;
  final String variantName;
  final int quantity;
  final double price;
  final bool isVeg;
  final String notes;
}

/// Drop-off address snapshot — captured on the order at checkout time, so it
/// stays correct even if the customer later edits/deletes the saved address.
class DeliveryAddressModel {
  DeliveryAddressModel({
    required this.label,
    required this.recipientName,
    required this.street,
    required this.additionalDetails,
    required this.city,
    required this.state,
    required this.zipCode,
    required this.phone,
  });

  factory DeliveryAddressModel.fromJson(Map<String, dynamic> json) {
    return DeliveryAddressModel(
      label: (json['label'] ?? '').toString(),
      recipientName: (json['fullName'] ?? json['name'] ?? '').toString(),
      street: (json['street'] ?? '').toString(),
      additionalDetails: (json['additionalDetails'] ?? '').toString(),
      city: (json['city'] ?? '').toString(),
      state: (json['state'] ?? '').toString(),
      zipCode: (json['zipCode'] ?? '').toString(),
      phone: (json['phone'] ?? '').toString(),
    );
  }

  final String label;
  final String recipientName;
  final String street;
  final String additionalDetails;
  final String city;
  final String state;
  final String zipCode;
  final String phone;

  bool get isEmpty => street.isEmpty && city.isEmpty;

  String get fullAddress => [
    street,
    additionalDetails,
    city,
    state,
    zipCode,
  ].where((s) => s.isNotEmpty).join(', ');
}

/// Full fare breakdown — mirrors `pricingSchema` on the backend order model.
class OrderPricingModel {
  OrderPricingModel({
    required this.subtotal,
    required this.tax,
    required this.packagingFee,
    required this.deliveryFee,
    required this.platformFee,
    required this.discount,
    required this.couponCode,
    required this.total,
  });

  factory OrderPricingModel.fromJson(Map<String, dynamic> json) {
    num? asNum(dynamic v) => v is num ? v : num.tryParse((v ?? '').toString());
    return OrderPricingModel(
      subtotal: asNum(json['subtotal'])?.toDouble() ?? 0,
      tax: asNum(json['tax'])?.toDouble() ?? 0,
      packagingFee: asNum(json['packagingFee'])?.toDouble() ?? 0,
      deliveryFee: asNum(json['deliveryFee'])?.toDouble() ?? 0,
      platformFee: asNum(json['platformFee'])?.toDouble() ?? 0,
      discount: asNum(json['discount'])?.toDouble() ?? 0,
      couponCode: (json['couponCode'] ?? '').toString(),
      total: asNum(json['total'])?.toDouble() ?? 0,
    );
  }

  final double subtotal;
  final double tax;
  final double packagingFee;
  final double deliveryFee;
  final double platformFee;
  final double discount;
  final String couponCode;
  final double total;
}

class OrderModel {
  OrderModel({
    required this.id,
    required this.displayId,
    required this.customerName,
    required this.customerPhone,
    required this.items,
    required this.deliveryAddress,
    required this.pricing,
    required this.total,
    required this.paymentMethod,
    required this.paymentStatus,
    required this.orderStatus,
    required this.dispatchStatus,
    required this.riderName,
    required this.riderPhone,
    required this.riderRating,
    required this.cancelledBy,
    required this.cancellationReason,
    required this.sendCutlery,
    required this.note,
    required this.deliveryInstructions,
    required this.acceptanceDeadlineAt,
    required this.createdAt,
  });

  factory OrderModel.fromJson(Map<String, dynamic> json) {
    num? asNum(dynamic v) => v is num ? v : num.tryParse((v ?? '').toString());
    final user = json['userId'];
    final userMap = user is Map ? user : null;
    final pricing = Map<String, dynamic>.from((json['pricing'] ?? {}) as Map);
    final payment = Map<String, dynamic>.from((json['payment'] ?? {}) as Map);
    final dispatch = Map<String, dynamic>.from((json['dispatch'] ?? {}) as Map);
    final deliveryAddress = Map<String, dynamic>.from(
      (json['deliveryAddress'] ?? {}) as Map,
    );
    final rider = dispatch['deliveryPartnerId'];
    final riderMap = rider is Map ? rider : null;
    final items = (json['items'] as List? ?? [])
        .map(
          (e) => OrderItemModel.fromJson(Map<String, dynamic>.from(e as Map)),
        )
        .toList();

    return OrderModel(
      id: (json['_id'] ?? json['orderMongoId'] ?? '').toString(),
      displayId: (json['order_id'] ?? json['orderId'] ?? '').toString(),
      customerName: (json['customerName'] ?? userMap?['name'] ?? 'Customer')
          .toString(),
      customerPhone: (json['customerPhone'] ?? userMap?['phone'] ?? '')
          .toString(),
      items: items,
      deliveryAddress: DeliveryAddressModel.fromJson(deliveryAddress),
      pricing: OrderPricingModel.fromJson(pricing),
      total: asNum(pricing['total'])?.toDouble() ?? 0,
      paymentMethod: (payment['method'] ?? '').toString(),
      paymentStatus: (payment['status'] ?? '').toString(),
      orderStatus: (json['orderStatus'] ?? json['status'] ?? '').toString(),
      dispatchStatus: (dispatch['status'] ?? 'unassigned').toString(),
      riderName: (riderMap?['name'] ?? riderMap?['fullName'] ?? '').toString(),
      riderPhone: (riderMap?['phone'] ?? riderMap?['phoneNumber'] ?? '')
          .toString(),
      riderRating: riderMap != null
          ? asNum(riderMap['rating'])?.toDouble()
          : null,
      cancelledBy: (json['cancelledBy'] ?? '').toString(),
      cancellationReason: (json['cancellationReason'] ?? '').toString(),
      sendCutlery: json['sendCutlery'] == true,
      note: (json['note'] ?? '').toString(),
      deliveryInstructions: (json['deliveryInstructions'] ?? '').toString(),
      acceptanceDeadlineAt: DateTime.tryParse(
        (json['acceptanceDeadlineAt'] ?? '').toString(),
      ),
      createdAt:
          DateTime.tryParse((json['createdAt'] ?? '').toString()) ??
          DateTime.now(),
    );
  }

  final String id;
  final String displayId;
  final String customerName;
  final String customerPhone;
  final List<OrderItemModel> items;
  final DeliveryAddressModel deliveryAddress;
  final OrderPricingModel pricing;
  final double total;
  final String paymentMethod;
  final String paymentStatus;
  final String orderStatus;
  final String dispatchStatus;
  final String riderName;
  final String riderPhone;
  final double? riderRating;
  final String cancelledBy;
  final String cancellationReason;
  final bool sendCutlery;
  final String note;
  final String deliveryInstructions;
  final DateTime? acceptanceDeadlineAt;
  final DateTime createdAt;

  bool get isAllVeg => items.isNotEmpty && items.every((item) => item.isVeg);
  bool get isCancelled => orderStatus.startsWith('cancelled');
  bool get hasRider => riderName.isNotEmpty;

  /// Restaurant-facing lifecycle bucket — matches the 7 tabs on the Orders screen.
  String get restaurantBucket {
    if (isCancelled) return 'cancelled';
    switch (orderStatus) {
      case 'created':
      case 'confirmed':
        return 'new';
      case 'preparing':
        return 'preparing';
      case 'ready_for_pickup':
        return 'ready';
      case 'reached_pickup':
      case 'picked_up':
      case 'reached_drop':
        return 'out_for_delivery';
      case 'delivered':
        return 'completed';
      default:
        return 'other';
    }
  }
}

File: backend/maava-new/Backend/src/modules/food/orders/models/order.model.js
import mongoose from 'mongoose';
import { verticalPlugin } from '../../../../core/vertical/verticalScope.js';

const orderItemSchema = new mongoose.Schema(
    {
        itemId: { type: String, required: true, trim: true },
        name: { type: String, required: true, trim: true },
        variantId: { type: String, trim: true, default: '' },
        variantName: { type: String, trim: true, default: '' },
        variantPrice: { type: Number, min: 0, default: 0 },
        price: { type: Number, required: true, min: 0 },
        /** Compare-at / other-platform unit price snapshot at order time. */
        otherPrice: { type: Number, min: 0, default: 0 },
        quantity: { type: Number, required: true, min: 1 },
        isVeg: { type: Boolean, default: true },
        /**
         * Rate this line was taxed at, snapshotted like the price is: a
         * product's GST slab can be reclassified, and the invoice has to keep
         * saying what was actually charged. null means the order-wide rate.
         */
        gstRate: { type: Number, min: 0, max: 100, default: null },
        brand: { type: String, trim: true, default: '' },
        packSize: { type: String, trim: true, default: '' },
        /**
         * Category this line belonged to, snapshotted at order time.
         *
         * Without it, "top selling categories" had to join order lines back to
         * the live catalogue by product name -- which silently drops every
         * product since renamed or deleted, and mis-attributes any two sellers
         * that use the same name. Recorded like the price is, and for the same
         * reason: the report must describe what was sold, not what the
         * catalogue happens to say today.
         */
        categoryId: { type: mongoose.Schema.Types.ObjectId, default: null },
        categoryName: { type: String, trim: true, default: '' },
        image: { type: String, default: '' },
        notes: { type: String, default: '' },
        /**
         * Add-ons chosen for this line, priced and named as at order time.
         *
         * Recorded rather than derived: an add-on's price can change, and the
         * order must keep what the customer was actually charged. `price` here
         * is per unit of the line, already folded into `price` above.
         *
         * The field did not exist before, so add-ons the customer selected were
         * dropped entirely — not billed, not shown, not recoverable afterwards.
         */
        addons: {
            type: [
                new mongoose.Schema(
                    {
                        addonId: { type: String, trim: true, default: '' },
                        name: { type: String, trim: true, default: '' },
                        price: { type: Number, min: 0, default: 0 }
                    },
                    { _id: false }
                )
            ],
            default: []
        }
    },
    { _id: false }
);

const deliveryAddressSchema = new mongoose.Schema(
    {
        label: { type: String, enum: ['Home', 'Office', 'Other'], default: 'Home' },
        name: { type: String, default: '', trim: true },
        fullName: { type: String, default: '', trim: true },
        street: { type: String, required: true, trim: true },
        additionalDetails: { type: String, default: '', trim: true },
        city: { type: String, required: true, trim: true },
        state: { type: String, required: true, trim: true },
        zipCode: { type: String, default: '', trim: true },
        phone: { type: String, default: '', trim: true },
        location: {
            type: { type: String, enum: ['Point'], default: 'Point' },
            coordinates: { type: [Number], default: undefined }
        }
    },
    { _id: false }
);

const pricingSchema = new mongoose.Schema(
    {
        subtotal: { type: Number, required: true, min: 0 },
        tax: { type: Number, default: 0, min: 0 },
        packagingFee: { type: Number, default: 0, min: 0 },
        deliveryFee: { type: Number, default: 0, min: 0 },
        deliveryFeeGst: { type: Number, default: 0, min: 0 },
        platformFee: { type: Number, default: 0, min: 0 },
        /** Extra surcharge when user selects Quick Mode (also included in platformFee). */
        quickDeliveryFee: { type: Number, default: 0, min: 0 },
        deliveryMode: { type: String, enum: ['basic', 'quick'], default: 'basic' },
        restaurantCommission: { type: Number, default: 0, min: 0 },
        discount: { type: Number, default: 0, min: 0 },
        /** Voluntary tip, paid straight through to the delivery partner. */
        deliveryTip: { type: Number, default: 0, min: 0 },
        couponCode: { type: String, default: null, trim: true, uppercase: true },
        total: { type: Number, required: true, min: 0 },
        currency: { type: String, default: 'INR' },
        /** Straight-line restaurant ↔ customer km (fee calculation) */
        distanceKm: { type: Number, default: null, min: 0 },
        /** Driving / road restaurant ↔ customer km (Directions API) */
        roadDistanceKm: { type: Number, default: null, min: 0 },
        roadDurationMins: { type: Number, default: null, min: 0 },
        /**
         * Packing minutes this order was quoted with, snapshotted like the
         * prices are. Per-vertical now, and an admin changing it must not
         * retroactively alter what an already-placed order promised.
         * null means "whatever the current default is" — every order created
         * before this field existed.
         */
        packingMinutes: { type: Number, default: null, min: 0 },
    },
    { _id: false }
);

const paymentSchema = new mongoose.Schema(
    {
        method: {
            type: String,
            enum: ['cash', 'razorpay', 'razorpay_qr', 'wallet'],
            required: true
        },
        status: {
            type: String,
            enum: [
                'cod_pending',
                'created',
                'authorized',
                'paid',
                'failed',
                'refunded',
                'pending_qr'
            ],
            default: 'cod_pending'
        },
        amountDue: { type: Number, min: 0 },
        razorpay: {
            orderId: { type: String },
            paymentId: { type: String },
            signature: { type: String }
        },
        qr: {
            qrId: { type: String },
            imageUrl: { type: String },
            paymentLinkId: { type: String },
            shortUrl: { type: String },
            status: { type: String },
            expiresAt: { type: Date }
        },
        // ✅ NEW: Added refund object to track refund status without breaking existing flow
        refund: {
            status: { 
                type: String, 
                enum: ['none', 'pending', 'processed', 'failed'], 
                default: 'none' 
            },
            amount: { type: Number, default: 0 },
            refundId: { type: String, default: '' },
            processedAt: { type: Date }
        }
    },
    { _id: false }
);

const dispatchSchema = new mongoose.Schema(
    {
        modeAtCreation: { type: String, enum: ['auto'], default: 'auto' },
        status: {
            type: String,
            enum: ['unassigned', 'assigned', 'accepted', 'rejected', 'cancelled'],
            default: 'unassigned'
        },
        deliveryPartnerId: { type: mongoose.Schema.Types.ObjectId, ref: 'FoodDeliveryPartner', default: null },
        assignedAt: { type: Date },
        acceptedAt: { type: Date },
        /** List of partners who were offered this order (to avoid repeats and track timeouts) */
        offeredTo: [{
            partnerId: { type: mongoose.Schema.Types.ObjectId, ref: 'FoodDeliveryPartner' },
            at: { type: Date, default: Date.now },
            action: { type: String, enum: ['offered', 'rejected', 'timeout', 'deassigned'], default: 'offered' }
        }],
        dispatchingAt: { type: Date }
    },
    { _id: false }
);

const deliveryStateSchema = new mongoose.Schema(
    {
        currentPhase: {
            type: String,
            enum: [
                'en_route_to_pickup',
                'at_pickup',
                'en_route_to_delivery',
                'at_drop',
                'delivered',
                'completed'
            ],
            default: 'en_route_to_pickup'
        },
        status: { type: String, default: '' },
        reachedPickupAt: { type: Date, default: null },
        reachedDropAt: { type: Date, default: null },
        pickedUpAt: { type: Date, default: null },
        deliveredAt: { type: Date, default: null }
    },
    { _id: false }
);

const statusHistorySchema = new mongoose.Schema(
    {
        at: { type: Date, default: Date.now },
        byRole: { type: String, enum: ['USER', 'RESTAURANT', 'DELIVERY_PARTNER', 'ADMIN', 'SYSTEM'] },
        byId: { type: mongoose.Schema.Types.ObjectId },
        from: { type: String },
        to: { type: String },
        note: { type: String, default: '' }
    },
    { _id: false }
);

const orderEntityRatingSchema = new mongoose.Schema(
    {
        rating: { type: Number, min: 1, max: 5 },
        comment: { type: String, default: '', trim: true },
        ratedAt: { type: Date, default: Date.now }
    },
    { _id: false }
);

/** One rated dish. itemId matches items[].itemId on the same order. */
const orderItemRatingSchema = new mongoose.Schema(
    {
        itemId: { type: String, required: true, trim: true },
        name: { type: String, default: '', trim: true },
        rating: { type: Number, min: 1, max: 5, required: true },
        comment: { type: String, default: '', trim: true },
        ratedAt: { type: Date, default: Date.now }
    },
    { _id: false }
);

const orderRatingsSchema = new mongoose.Schema(
    {
        restaurant: { type: orderEntityRatingSchema, default: undefined },
        deliveryPartner: { type: orderEntityRatingSchema, default: undefined },
        /** The CUSTOMER, rated by the delivery partner after handover. */
        customer: { type: orderEntityRatingSchema, default: undefined },
        /** Per-dish ratings from the customer. */
        items: { type: [orderItemRatingSchema], default: [] }
    },
    { _id: false }
);

const deliveryVerificationSchema = new mongoose.Schema(
    {
        dropOtp: {
            required: { type: Boolean, default: false },
            verified: { type: Boolean, default: false }
        }
    },
    { _id: false }
);

const orderSchema = new mongoose.Schema(
    {
        order_id: {
            type: String,
            unique: true,
            sparse: true,
            index: true
        },
        /** Compatibility alias: satisfies rogue unique index 'orderId_1' found in legacy deployments. */
        orderId: {
            type: String,
            unique: true,
            sparse: true,
            index: true
        },
        userId: {
            type: mongoose.Schema.Types.ObjectId,
            ref: 'FoodUser',
            required: true
        },
        restaurantId: {
            type: mongoose.Schema.Types.ObjectId,
            ref: 'FoodRestaurant',
            required: true
        },
        zoneId: {
            type: mongoose.Schema.Types.ObjectId,
            ref: 'FoodZone',
            index: true
        },
        transactionId: {
            type: mongoose.Schema.Types.ObjectId,
            ref: 'FoodTransaction',
            index: true
        },
        items: {
            type: [orderItemSchema],
            required: true,
            validate: (v) => Array.isArray(v) && v.length > 0
        },
        deliveryAddress: {
            type: deliveryAddressSchema,
            required: true
        },
        customerName: { type: String, default: '', trim: true },
        customerPhone: { type: String, default: '', trim: true },
        pricing: {
            type: pricingSchema,
            required: false
        },
        /**
         * Denormalized payment snapshot for fast reads & legacy clients.
         * Authoritative audit trail: collection `food_order_payments` (FoodOrderPayment model).
         */
        payment: {
            type: paymentSchema,
            required: false
        },
        orderStatus: {
            type: String,
            enum: [
                'pending_payment',
                'created',
                'confirmed',
                'preparing',
                'ready_for_pickup',
                'reached_pickup',
                'picked_up',
                'reached_drop',
                'delivered',
                'cancelled_by_user',
                'cancelled_by_restaurant',
                'cancelled_by_admin'
            ],
            default: 'created'
        },
        dispatch: {
            type: dispatchSchema,
            default: () => ({})
        },
        deliveryState: {
            type: deliveryStateSchema,
            default: () => ({})
        },
        statusHistory: {
            type: [statusHistorySchema],
            default: []
        },
        ratings: {
            type: orderRatingsSchema,
            default: () => ({})
        },
        note: { type: String, default: '', trim: true },
        deliveryInstructions: { type: String, default: '', trim: true },
        acceptanceWindowSeconds: { type: Number, default: 240, min: 1 },
        acceptanceDeadlineAt: { type: Date, default: null },
        /** Idempotency guard so retries/duplicate calls never double-push the "new order" alert. */
        restaurantNotifiedAt: { type: Date, default: null },
        /** Set once stock was decremented for this order; absent on pre-inventory orders. */
        stockReservedAt: { type: Date, default: null },
        /**
         * Set once stock was given back. Guards the restock, which is reachable
         * from user cancel, seller cancel, admin cancel, the acceptance-timeout
         * sweep and two delete paths — several of which can race. Restocking
         * twice silently inflates inventory, and nothing downstream would notice.
         */
        stockRestoredAt: { type: Date, default: null },
        sendCutlery: { type: Boolean, default: true },
        deliveryFleet: { type: String, default: 'standard', trim: true },
        scheduledAt: { type: Date, default: null },
        riderEarning: { type: Number, default: 0, min: 0 },
        // Can be negative when discounts/rider pay exceed platform income; keep the real value visible.
        platformProfit: { type: Number, default: 0 },
        /** Restaurant ↔ customer driving distance (km) for delivery-partner offer UI */
        tripDistanceKm: { type: Number, default: null, min: 0 },
        tripDurationMins: { type: Number, default: null, min: 0 },
        /** Plain 4-digit OTP for handover; cleared after successful verify (never expose to partner in API responses). */
        deliveryOtp: { type: String, default: '', select: false },
        deliveryVerification: {
            type: deliveryVerificationSchema,
            default: () => ({})
        },
        /** Latest rider location for this specific order (GeoJSON Point) */
        lastRiderLocation: {
            type: { type: String, enum: ['Point'] },
            coordinates: { type: [Number] }
        }
    },
    {
        collection: 'food_orders',
        timestamps: true
    }
);

orderSchema.plugin(verticalPlugin);

/**
 * `vertical` leads every compound index below.
 *
 * Once both verticals share one collection, an index that does not lead with it
 * makes each of these queries scan the other vertical's orders as well -- the
 * collection doubles and so does the work, for a filter every query now carries.
 * Built now, while each collection is still half its eventual size, because a
 * rebuild after the data merge is the expensive time to do it.
 */
orderSchema.index({ vertical: 1, createdAt: -1 });
orderSchema.index({ vertical: 1, orderStatus: 1, createdAt: -1 });
orderSchema.index({ 'deliveryAddress.location': '2dsphere' });
orderSchema.index({ lastRiderLocation: '2dsphere' });
orderSchema.index({ vertical: 1, userId: 1, createdAt: -1 });
/**
 * Kept WITHOUT vertical as well, deliberately. "All my orders, both verticals"
 * is a real query the moment one customer shops in both -- it is most of the
 * reason for merging -- and it runs with skipVerticalScope, which the
 * vertical-leading index above cannot serve.
 */
orderSchema.index({ userId: 1, createdAt: -1 });
orderSchema.index({ vertical: 1, restaurantId: 1, orderStatus: 1, createdAt: -1 });
orderSchema.index({ vertical: 1, 'dispatch.deliveryPartnerId': 1, orderStatus: 1 });
orderSchema.index({ vertical: 1, 'dispatch.status': 1, orderStatus: 1 });
orderSchema.index({ vertical: 1, 'dispatch.status': 1, orderStatus: 1, updatedAt: -1 });
/**
 * Rider-facing, and NOT vertical-led: one rider takes a grocery run at 4pm and a
 * dinner order at 8pm off the same shared fleet, so their active-job lookup must
 * cross verticals by design.
 */
orderSchema.index({ 'dispatch.deliveryPartnerId': 1, 'dispatch.status': 1, updatedAt: -1 });
orderSchema.index({ vertical: 1, 'payment.status': 1, createdAt: -1 });
orderSchema.index({ vertical: 1, 'payment.method': 1, createdAt: -1 });

orderSchema.pre('save', async function (next) {
    try {
        if (!this.order_id) {
            // 6 timestamp digits + 4 random digits, verified against the collection.
            // The old 4+3 format collided after a few thousand orders (birthday paradox),
            // which made display-id lookups match the wrong order.
            for (let attempt = 0; attempt < 5 && !this.order_id; attempt += 1) {
                const timestamp = Date.now().toString().slice(-6);
                const random = Math.floor(1000 + Math.random() * 9000);
                const candidate = `FOD-${timestamp}${random}`;
                const exists = await this.constructor.exists({
                    $or: [{ order_id: candidate }, { orderId: candidate }],
                });
                if (!exists) this.order_id = candidate;
            }
            if (!this.order_id) {
                // Guaranteed unique: derived from this document's own ObjectId.
                this.order_id = `FOD-${this._id.toString().slice(-10).toUpperCase()}`;
            }
        }
        // Synchronize camelCase alias to satisfy unique index 'orderId_1'
        if (this.order_id) {
            this.orderId = this.order_id;
        }
        next();
    } catch (err) {
        next(err);
    }
});

export const FoodOrder = mongoose.model('FoodOrder', orderSchema);

const settingsSchema = new mongoose.Schema(
    {
        key: { type: String, required: true, trim: true },
        dispatchMode: { type: String, enum: ['auto'], default: 'auto' },
        updatedBy: {
            role: { type: String },
            adminId: { type: mongoose.Schema.Types.ObjectId },
            at: { type: Date }
        }
    },
    { collection: 'food_settings', timestamps: true }
);

settingsSchema.plugin(verticalPlugin);

/**
 * ponytail: the legacy unique index `key_1` still exists on deployed
 * clusters and must be dropped before the two databases merge -- until then
 * it rejects the second vertical's copy of any given key. Nothing collides
 * while each database holds one vertical, so the drop belongs with the phase 5
 * migration.
 */
settingsSchema.index({ vertical: 1, key: 1 }, { unique: true });

export const FoodSettings = mongoose.model('FoodSettings', settingsSchema);
